translate/storage/ts2.py

```diff
@@ -488,6 +489,20 @@ def serialize(self, out):
         # Qt Linguist does self-close the "location" elements though
         for e in root.xpath("//*[not(./node()) and not(text()) and not(name() = 'location')]"):
             e.text = ""
+        # For conformance with Qt output, write XML declaration with double quotes
         out.write(str('<?xml version="1.0" encoding="utf-8"?>\n').encode('utf-8'))
-        out.write(etree.tostring(root, doctype=doctype, pretty_print=True,
-                                 xml_declaration=None, encoding='utf-8'))
+        # For conformance with Qt output, post-process etree.tostring output,
+        # replacing ' with &apos; and " with &quot; in text elements
+        treestring = etree.tostring(root, doctype=doctype, pretty_print=True,
+                                    xml_declaration=False, encoding='utf-8')
+        treestring = codecs.decode(treestring, 'utf-8')
```

Comment:
You should use encoding='unicode' for etree.tostring(), that saves need for the additional utf-8 decoding.